controlplane/kubeadm/internal/workload_cluster_etcd.go

```diff
@@ -130,6 +137,39 @@ func (w *Workload) EtcdIsHealthy(ctx context.Context) (HealthCheckResult, error)
 		return response, errors.Errorf("there are %d healthy etcd pods, but %d etcd members", expectedMembers, len(knownMemberIDSet))
 	}
 
+	// Check etcd cluster alarms
+	etcdClient, err := w.etcdClientGenerator.forNodes(ctx, controlPlaneNodes.Items)
+	if err != nil {
+		return response, err
+	}
+
+	defer etcdClient.Close()
+	alarmList, err := etcdClient.Alarms(ctx)
+	if len(alarmList) > 0 || err != nil {
+		return response, errors.Errorf("etcd cluster has %d alarms", len(alarmList))
+	}
+
+	members, err := etcdClient.Members(ctx)
+	if err != nil {
+		return response, err
+	}
+
+	healthyMembers := 0
+	for _, m := range members {
+		if val, ok := response[m.Name]; ok {
+			if val == nil {
+				healthyMembers++
+			}
+		} else {
+			// There are members in etcd cluster that is not part of controlplane nodes.
+			return response, err
```

Comment:
why are we returning err from L152, should this be a new err?

Reply:
Yes, also instead of returning errors here, I accumulate them in a list and return at the end now.